quantumview-template.tex

```diff
@@ -0,0 +1,66 @@
+
+Quantum provides the quantumview documentclass to enable authors of Views to use their common LaTeX environment to prepare their contributions. The editors at Quantum can then generate the HTML output by supplying the \texttt{html} option.
+
+\section{Supported Formatting Options}
+The documentclass natively supports the following operations:
+
+\paragraph{Text formatting} The following text formats are supported: \emph{emphasis}, \textit{italic}, \textbf{bold}, \texttt{typewriter}, \textsuperscript{superscript} and \textsubscript{subscript}.
+
+\paragraph{Sectioning} Sectioning -- if needed -- can be performed using the regular \texttt{{\textbackslash}section}, \texttt{{\textbackslash}subsection}, \texttt{{\textbackslash}subsubsection} and \texttt{{\textbackslash}paragraph} commands. These will be converted to HTML header tags and therefore not show section numbers in the final HTML.
+
+\paragraph{Citations and Bibliography} You can cite references using the regular \texttt{{\textbackslash}cite} command. For example, here is some text citing a textbook~\cite{Nielsen_Chuang_2000}, a journal article~\cite{Preskill2018}, a newer preprint~\cite{SchwarzhansLockErkerFriisHuber2020} and a journal article whose preprint has an arXiv identifier in old format~\cite{AcinBrussLewensteinSanpera2001}.
+
+Please see quantumview-template.bib for an example of how to provide bibliographic information to BibLaTeX in a way that yields a suitable bibliography with DOI links.
+In both Quantum and Quantum Views all citations to cited works that have a DOI must include a hyperlink to the DOI of the work.
+
+\paragraph{Formulas} You are free to use inline math $\mathcal{Z} - \pi = \nabla \Gamma$ and both the \texttt{equation}
+\begin{equation}
+    \int_0^1 \mathrm{d}x \, |\psi(x)\rangle \! \langle \psi(x)|= \hat{O}^2
+\end{equation}
+and \texttt{align} environment
+\begin{align}
+    \oint_C = \mathcal{Z}^2.
+\end{align}
+As formulas are directly rendered on the webpage, \emph{you can not use custom commands and libraries}. If you are unsure wether or not the command you want to use is supported, please consult the MathJax documentation. You should thus refrain from using the \texttt{{\textbackslash}label} and \texttt{{\textbackslash}ref} commands.
+
+\paragraph{Lists} You are free to use both \texttt{itemize} for unordered lists,
+\begin{itemize}
+    \item Item 1 lorem ipsum
+    \item Item 2
+\end{itemize}
+and \texttt{enumerate} for ordered lists:
+\begin{enumerate}
+    \item Item 1
+    \item Item 2
+\end{enumerate}
+Note that further modifiers, \emph{e.g.}\ for roman numbering and additional packages like \texttt{enumerate} are not supported.
+
+\section{Copy-Editing Tools}
+The quantumview document class also provides commands that are useful in copy-editing. These are \texttt{{\textbackslash}corr} for \corr{correctons}{corrections} and \texttt{{\textbackslash}ins} \ins{for insertions}.
+
+\bibliography{quantumview-template}
+\bibliographystyle{unsrtnat}
```

Comment:
I would like to switch that to `\bibliographystyle{quantum}` but when doing this and running BibTeX I get:
```
Running `BibTeX' on `quantumview-template' with ``bibtex quantumview-template''
This is BibTeX, Version 0.99d (TeX Live 2016/Debian)
The top-level auxiliary file: quantumview-template.aux
The style file: quantum.bst
Database file #1: quantumview-template.bib
Warning--empty doi, eprint and url in Nielsen_Chuang_2000
You can't pop an empty literal stack for entry Nielsen_Chuang_2000
while executing---line 1641 of file quantum.bst
(There was 1 error message)
```
Does that work for you @dwierichs ? Maybe my BibTeX is too old?

Reply:
Nope, this is supposed to crash since the entry Nielsen_Chuang_2000 does not contain any link and the bibstyle enforces at least one link to be present. I'll add the DOI to that entry, then it works for me.

Reply:
Ok, I see. There is the warning in the BibTeX output about doi, eprint, and url beeing empty, but it is only a warning and thus I expected it to work despite the warning. To an innocent user the "can't pop..." line looks more like a bug/crash than a informative error message. Can we make bibtex say something like "One of doi, eprint, or url must be provided? What about old books that may genuinely not have any of those three?

Reply:
Essentially the warning "empty doi, eprint and url in [entry]" is the warning you mean that tells the user that one of them is needed. We have not thought about entries for which none of the three is available.

The expected behaviour (and the one I get locally) here would be that the compilation actually crashes, because of our decision against silent bibtex problems and the bibstyle turning warnings into errors. However, it should not complain about the stack being empty, but simply raise an error (again, which it does for me personally locally).